File: composeApp/src/androidMain/kotlin/com/example/ergoguard/ui/ResultScreen.kt
package com.example.ergoguard.ui

import android.app.Activity
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.SideEffect
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.toArgb
import androidx.compose.ui.platform.LocalView
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.core.view.WindowCompat
import com.example.ergoguard.R
import com.example.ergoguard.model.AnalysisResult
import com.example.ergoguard.model.ComparisonResult
import com.example.ergoguard.model.SeverityLevel

/**
 * Màn hình hiển thị kết quả phân tích tư thế.
 */
@Composable
fun ResultScreen(
    result: AnalysisResult,
    comparison: ComparisonResult?,
    isFirstCapture: Boolean,
    onRetakeClick: () -> Unit,
    onResetClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    val view = LocalView.current
    val isDarkTheme = isSystemInDarkTheme()
    val backgroundColor = MaterialTheme.colorScheme.background
    if (!view.isInEditMode) {
        SideEffect {
            val window = (view.context as Activity).window
            window.statusBarColor = backgroundColor.toArgb()
            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = !isDarkTheme
        }
    }
    val levelColor = when (result.level) {
        SeverityLevel.NORMAL -> MaterialTheme.colorScheme.tertiary // Emerald/Green from M3 palette
        SeverityLevel.WARNING -> Color(0xFFB98200) // Darker warning for M3 contrast
        SeverityLevel.DANGER -> MaterialTheme.colorScheme.error
    }

    val statusText = when (result.level) {
        SeverityLevel.NORMAL -> stringResource(R.string.status_safe)
        SeverityLevel.WARNING -> stringResource(R.string.status_warning_mild)
        SeverityLevel.DANGER -> stringResource(R.string.status_warning_severe)
    }

    Scaffold(
        modifier = modifier.fillMaxSize(),
        containerColor = MaterialTheme.colorScheme.background
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(16.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(modifier = Modifier.height(32.dp))

            // Percentage display - Standard Text
            Column(
                modifier = Modifier.wrapContentSize(),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Text(
                    text = "${result.protrusionPercentage.toInt()}%",
                    style = MaterialTheme.typography.displayLarge,
                    color = levelColor,
                    fontWeight = FontWeight.Bold
                )
                Text(
                    text = statusText,
                    style = MaterialTheme.typography.headlineMedium,
                    color = levelColor
                )
            }

            Spacer(modifier = Modifier.height(24.dp))

            // Neck load - Standard Card with Shadow
            Card(
                modifier = Modifier.fillMaxWidth(),
                elevation = CardDefaults.cardElevation(defaultElevation = 4.dp),
                colors = CardDefaults.cardColors(
                    containerColor = MaterialTheme.colorScheme.surfaceVariant,
                    contentColor = MaterialTheme.colorScheme.onSurfaceVariant
                )
            ) {
                Column(
                    modifier = Modifier
                        .padding(16.dp)
                        .fillMaxWidth(),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Text(
                        text = stringResource(R.string.result_neck_load),
                        style = MaterialTheme.typography.titleMedium
                    )
                    Text(
                        text = "~${result.neckLoadKg.toInt()} kg",
                        style = MaterialTheme.typography.displayMedium,
                        color = levelColor,
                        fontWeight = FontWeight.Bold
                    )
                }
            }

            Spacer(modifier = Modifier.height(16.dp))

            // Message
            Text(
                text = result.message,
                style = MaterialTheme.typography.bodyLarge,
                textAlign = TextAlign.Center,
                color = MaterialTheme.colorScheme.onBackground,
                modifier = Modifier.padding(horizontal = 16.dp)
            )

            // Comparison result (if not first capture)
            if (comparison != null) {
                Spacer(modifier = Modifier.height(16.dp))
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
                    colors = CardDefaults.cardColors(
                        containerColor = if (comparison.isImproved)
                            MaterialTheme.colorScheme.tertiaryContainer
                        else
                            MaterialTheme.colorScheme.errorContainer
                    )
                ) {
                    Text(
                        text = comparison.improvementMessage,
                        modifier = Modifier
                            .padding(16.dp)
                            .fillMaxWidth(),
                        style = MaterialTheme.typography.bodyLarge,
                        textAlign = TextAlign.Center,
                        fontWeight = FontWeight.Medium,
                        color = if (comparison.isImproved)
                            MaterialTheme.colorScheme.onTertiaryContainer
                        else
                            MaterialTheme.colorScheme.onErrorContainer
                    )
                }
            }

            Spacer(modifier = Modifier.height(24.dp))

            // Fix Action
            Card(
                modifier = Modifier.fillMaxWidth(),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
                colors = CardDefaults.cardColors(
                    containerColor = MaterialTheme.colorScheme.secondaryContainer,
                    contentColor = MaterialTheme.colorScheme.onSecondaryContainer
                )
            ) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Text(
                        text = stringResource(R.string.result_exercise_title),
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold
                    )
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        text = result.fixAction,
                        style = MaterialTheme.typography.bodyMedium
                    )
                }
            }

            Spacer(modifier = Modifier.weight(1f))

            // Buttons
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                OutlinedButton(
                    onClick = onResetClick,
                    modifier = Modifier.weight(1f)
                ) {
                    Text(stringResource(R.string.btn_start_over))
                }

                Button(
                    onClick = onRetakeClick,
                    modifier = Modifier.weight(1f)
                ) {
                    Text(
                        if (isFirstCapture) stringResource(R.string.btn_retake)
                        else stringResource(R.string.btn_compare)
                    )
                }
            }

            Spacer(modifier = Modifier.height(16.dp))
        }
    }
}

@androidx.compose.ui.tooling.preview.Preview(showBackground = true)
@Composable
private fun ResultScreenPreview() {
    com.example.ergoguard.ui.theme.ErgoGuardTheme {
        ResultScreen(
            result = AnalysisResult.forSideView(
                protrusionPercentage = 25.0,
                level = SeverityLevel.WARNING,
                neckLoadKg = 12.0,
                message = "Cơ cổ đang phải gánh gấp đôi trọng lượng đầu.",
                fixAction = "Giữ đầu thẳng, dùng ngón tay đẩy nhẹ cằm về phía sau."
            ),
            comparison = null,
            isFirstCapture = true,
            onRetakeClick = {},
            onResetClick = {}
        )
    }
}

File: iosApp/iosApp/UI/Screens/ResultScreen.swift
import SwiftUI

// MARK: - Result Screen
struct ResultScreen: View {
    let result: AnalysisResult
    let comparison: ComparisonResult?
    let isFirstCapture: Bool
    let onRetakeClick: () -> Void
    let onResetClick: () -> Void

    @Environment(\.colorScheme) var colorScheme

    private var backgroundColor: Color {
        colorScheme == .dark ? ErgoGuardColors.Dark.background : ErgoGuardColors.background
    }

    private var textColor: Color {
        colorScheme == .dark ? ErgoGuardColors.Dark.onBackground : ErgoGuardColors.onBackground
    }

    private var surfaceColor: Color {
        colorScheme == .dark ? ErgoGuardColors.Dark.surfaceVariant : ErgoGuardColors.surfaceVariant
    }

    var body: some View {
        ZStack {
            backgroundColor
                .ignoresSafeArea()

            ScrollView {
                VStack(spacing: 24) {
                    Spacer()
                        .frame(height: 32)

                    // Main percentage display
                    percentageDisplay

                    // Neck load card
                    neckLoadCard

                    // Message
                    messageSection

                    // Comparison result
                    if let comparison = comparison {
                        comparisonCard(comparison)
                    }

                    // Fix action card
                    fixActionCard

                    Spacer()
                        .frame(height: 24)

                    // Action buttons
                    actionButtons

                    Spacer()
                        .frame(height: 16)
                }
                .padding(.horizontal, 16)
            }
        }
    }

    // MARK: - Percentage Display
    private var percentageDisplay: some View {
        VStack(spacing: 8) {
            Text("\(Int(result.protrusionPercentage))%")
                .font(ErgoGuardTypography.displayMedium)
                .foregroundColor(result.level.color)

            Text(result.level.label)
                .font(ErgoGuardTypography.headlineMedium)
                .foregroundColor(result.level.color)
        }
    }

    // MARK: - Neck Load Card
    private var neckLoadCard: some View {
        LiquidGlassCard(cornerRadius: 16) {
            VStack(spacing: 8) {
                Text("neck_load_title")
                    .font(ErgoGuardTypography.titleMedium)
                    .foregroundColor(textColor.opacity(0.8))

                Text("~\(Int(result.neckLoadKg)) kg")
                    .font(ErgoGuardTypography.displayMedium)
                    .foregroundColor(result.level.color)
            }
            .frame(maxWidth: .infinity)
            .padding(.vertical, 20)
            .padding(.horizontal, 16)
        }
        .background(
            RoundedRectangle(cornerRadius: 16)
                .fill(surfaceColor.opacity(0.5))
        )
    }

    // MARK: - Message Section
    private var messageSection: some View {
        Text(result.message)
            .font(ErgoGuardTypography.bodyLarge)
            .foregroundColor(textColor)
            .multilineTextAlignment(.center)
            .padding(.horizontal, 16)
    }

    // MARK: - Comparison Card
    private func comparisonCard(_ comparison: ComparisonResult) -> some View {
        LiquidGlassCard(cornerRadius: 12) {
            Text(comparison.improvementMessage)
                .font(ErgoGuardTypography.titleMedium)
                .foregroundColor(comparison.isImproved ? ErgoGuardColors.tertiary : ErgoGuardColors.error)
                .multilineTextAlignment(.center)
                .frame(maxWidth: .infinity)
                .padding(16)
        }
        .background(
            RoundedRectangle(cornerRadius: 12)
                .fill(comparison.isImproved
                      ? ErgoGuardColors.tertiaryContainer.opacity(0.3)
                      : ErgoGuardColors.errorContainer.opacity(0.3))
        )
    }

    // MARK: - Fix Action Card
    private var fixActionCard: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack(spacing: 8) {
                Image(systemName: "figure.strengthtraining.traditional")
                    .font(ErgoGuardTypography.titleMedium)
                    .foregroundColor(ErgoGuardColors.tertiary)

                Text("exercise_title")
                    .font(ErgoGuardTypography.bodyLarge)
                    .fontWeight(.bold)
                    .foregroundColor(textColor)
            }

            Text(result.fixAction)
                .font(ErgoGuardTypography.bodyMedium)
                .foregroundColor(textColor.opacity(0.8))
                .lineSpacing(4)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(16)
        .background(
            RoundedRectangle(cornerRadius: 16)
                .fill(ErgoGuardColors.secondaryContainer.opacity(colorScheme == .dark ? 0.2 : 0.5))
        )
        .overlay(
            RoundedRectangle(cornerRadius: 16)
                .stroke(ErgoGuardColors.secondaryContainer.opacity(0.3), lineWidth: 1)
        )
    }

    // MARK: - Action Buttons
    private var actionButtons: some View {
        HStack(spacing: 16) {
            GlassButton(
                title: "btn_reset",
                action: onResetClick,
                isPrimary: false
            )

            GlassButton(
                title: isFirstCapture ? "btn_retake" : "btn_compare",
                action: onRetakeClick
            )
        }
    }
}

// MARK: - Preview
#if DEBUG
struct ResultScreen_Previews: PreviewProvider {
    static var previews: some View {
        Group {
            ResultScreen(
                result: AnalysisResult(
                    protrusionPercentage: 25,
                    level: .warning,
                    neckLoadKg: 12,
                    message: "Cơ cổ đang phải gánh gấp đôi trọng lượng đầu.",
                    fixAction: "Giữ đầu thẳng, dùng ngón tay đẩy nhẹ cằm về phía sau. Giữ 10 giây, lặp lại 3 lần."
                ),
                comparison: nil,
                isFirstCapture: true,
                onRetakeClick: {},
                onResetClick: {}
            )
            .preferredColorScheme(.light)

            ResultScreen(
                result: AnalysisResult(
                    protrusionPercentage: 45,
                    level: .danger,
                    neckLoadKg: 22,
                    message: "Áp lực cổ nghiêm trọng! Cần điều chỉnh tư thế ngay.",
                    fixAction: "Thực hiện bài tập Chin Tuck: Đưa cằm về phía sau như đang tạo nọng."
                ),
                comparison: ComparisonResult(
                    percentageChange: -15,
                    isImproved: true,
                    improvementMessage: "Tuyệt vời! Bạn đã cải thiện 15% so với lần trước!"
                ),
                isFirstCapture: false,
                onRetakeClick: {},
                onResetClick: {}
            )
            .preferredColorScheme(.dark)
        }
    }
}
#endif
